src/Ng/Jobs/BackupV3Storage.cs

```diff
@@ -0,0 +1,236 @@
+                    p.Close();
+                }
+                catch (Exception ex)
+                {
+                    Logger.LogError($"BackupV3Storage:Exception on backup save.{ex.ToString()}.");
+                    return false;
+                }
+                try
+                {
+                    ProcessStartInfo copyToAzureProcLog = new ProcessStartInfo();
+                    copyToAzureProcLog.FileName = @"cmd.exe";
+                    copyToAzureProcLog.Arguments = $"/c {AzCopyPath} {argumentsLog}";
+                    copyToAzureProcLog.CreateNoWindow = false;
+                    copyToAzureProcLog.UseShellExecute = false;
+                    Process pLog = Process.Start(copyToAzureProcLog);
+                    pLog.WaitForExit();
+                    pLog.Close();
+                }
+                catch (Exception ex)
+                {
+                    Logger.LogError($"BackupV3Storage:Exception on log save.{ex.ToString()}.");
+                }
+
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private void RefreshLogData(string logFolder, string logFile)
+        {
+            if(Directory.Exists(logFolder))
+            {
+                Directory.Delete(logFolder, true);
+            }
+            Directory.CreateDirectory(logFolder);
+            var stream = File.Create(logFile);
+            stream.Close();
+        }
+
+        private async Task<bool> TryCreateDestinationContainerAsync(string containerName, string storageAccountName, string storageAccountKey)
+        {
+            CloudBlobClient blobClient = GetCloudBlobClient(storageAccountName, storageAccountKey);
+
+            try
+            {
+                CloudBlobContainer container = blobClient.GetContainerReference(containerName);
+                await container.CreateIfNotExistsAsync();
+                return true;
+            }
+            catch(Exception ex)
+            {
+                Logger.LogError($"BackupV3Storage:Exception on create container {containerName}.{ex.ToString()}.");
+                return false;
+            }
+        }
+
+        private async Task<bool> TryDeleteContainerAsync(string containerName, DateTime date, string storageAccountName, string storageAccountKey)
+        {
+            string destContainer = $"{Format(date.AddDays(-1* _backupDays))}-{containerName}";
+            CloudBlobClient blobClient = GetCloudBlobClient(storageAccountName, storageAccountKey);
+
+            try
+            {
+                CloudBlobContainer container = blobClient.GetContainerReference(destContainer);
+                await container.DeleteIfExistsAsync();
+                return true;
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError($"BackupV3Storage:Exception on delete container {destContainer}.{ex.ToString()}.");
```

Comment:
Log error has a special overload for an exception, right?

Also, does this end up in the `exceptions` table in AI?